### PR TITLE
Add templates.json

### PR DESCRIPTION
**Pull request by martindekov**
Adding templates.json file to be consumed by faas-cli
by listing the existing templates so people know what we support

Signed-off-by: Martin Dekov (VMware) <mdekov@vmware.com>

### CODE REVIEW

**Comment by rgee0** [comment]
Can we add an attribute for hardware platform, please?

Could we also have a pure language field?

**Comment by martindekov** [comment]
Yeah totally, open to suggestions, forgot to change title to [WIP]

**Comment by rgee0** [comment]
Hello @martindekov.  When I referred to 'pure language' I was meaning a field the held just the language, so that it is simple to filter at the client.  In the initial commit it looked as though we had a name field and a language field that were very close in value content, hence some redundancy.  I had something like this in mind when I saw your earlier proposal:
```
    {
        "name": "go-armhf",
        "platform": "armhf"
        "language": "go"
        "source": "openfaas",
        "description": "Official Golang on armhf template",
        "repo": "https://github.com/openfaas/templates",
        "official": "true"
    },
```

Or with a slight modification:
```
    {
        "cli-lang": "go-armhf",
        "platform": "armhf"
        "language": "go"
        "source": "openfaas",
        "description": "Official Golang on armhf template",
        "repo": "https://github.com/openfaas/templates",
        "official": "true"
    },
```

File: templates.json [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    },
+    {
+        "name": "ruby",
+        "source": "openfaas",
+        "description": "Official Ruby x86-64 template",
+        "repo": "https://github.com/openfaas/templates",
+        "official": "true"
+    },
+    {
+        "name": "node10-express-template",
```
Comment (alexellis) [inline templates.json]: This is the repo name. What is the name in the `template.yml` file? - I think it's `node10-express`

So I should be able to do `faas template store pull node10-express` or fully qualified: `faas template store pull openfaas-incubator/node10-express-template`

File: templates.json [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    },
+    {
+        "name": "node8-express-template",
+        "source": "openfaas-incubator",
+        "description": "NodeJS 8 Express template",
+        "repo": "https://github.com/openfaas-incubator/node8-express-template",
+        "official": "false"
+    },
+    {
+        "name": "golang-http-template",
```
Comment (alexellis) [inline templates.json]: There's two templates in this repo btw.

**Comment by alexellis** [comment]
As this list grows perhaps adding tags would be useful?

For example:

`"tags": [ "csharp", "c#", "dotnet" ]`

File: templates.json (hunk 1 above)
Comment (martindekov) [reply templates.json]: Yeah it is `node10-express` but not sure what you mean here, can we pull single folder from repository? Or you mean to read the template.yml and pull the whole repository if the single template that we need exists?

File: templates.json (hunk 2 above)
Comment (martindekov) [reply templates.json]: I know, do you mean I should add some array field which shows all templates inside this repository?

**Comment by alexellis** [comment]
This will mean some duplication such as a repo name, but I think that is probably ok for first pass.

**Comment by martindekov** [comment]
It is ready for review

File: templates.json (hunk 2 above)
Comment (martindekov) [reply templates.json]: This is now resolved thanks for the feedback on that one did not though of it in that way 🍻

File: templates.json (hunk 1 above)
Comment (martindekov) [reply templates.json]: This is now referenced ^

**Comment by martindekov** [comment]
I think this is ready to be merged? WDYT @rgee0 @alexellis

**Comment by rgee0** [comment]
LGTM

Note: this is required prior to merging openfaas/faas-cli#561

**Comment by alexellis** [comment]
There are a few things I'd like to change here, but I do believe in incremental progress so I'll merge now then let's revisit a few things. For one there is an inconsistency between the way we are managing JSON files for the function store platform vs template store. I also think `cli-lang` is a bit odd an arbitrary, why isn't this `lang` like the YAML file?

**Comment by rgee0** [comment]
The inconsistency between how we are managing JSON files between functions and templates is partly due to the surprise nature of the original store.  I tried to address this in https://github.com/openfaas/faas/issues/429.  If we were creating the store today I would be very surprised if we opted to maintain multiple files over a single file.

The CLI (openfaas/faas-cli#561) implementation also includes the inconsistency in that you interact differently with the CLI depending upon whether you want functions - `faas-cli store` - or templates - `faas-cli template store`.

**Pull request merged**